components/http_client.rst

```diff
@@ -0,0 +1,355 @@
+.. index::
+   single: HttpClient
+   single: Components; HttpClient
+
+The HttpClient Component
+========================
+
+    The HttpClient component is a low-level HTTP client with support for both
+    PHP stream wrappers and cURL. It also provides utilities to consume APIs.
+
+.. versionadded:: 4.3
+
+    The HttpClient component was introduced in Symfony 4.3.
+
+Installation
+------------
+
+.. code-block:: terminal
+
+    $ composer require symfony/http-client
+
+Alternatively, you can clone the `<https://github.com/symfony/http-client>`_
+repository.
+
+.. include:: /components/require_autoload.rst.inc
+
+Basic Usage
+-----------
+
+Use the :class:`Symfony\Component\HttpClient\HttpClient` class to create the
+low-level HTTP client that makes requests, like the following ``GET`` request::
+
+    use Symfony\Component\HttpClient\HttpClient;
+
+    $httpClient = HttpClient::create();
+    $response = $httpClient->request('GET', 'https://api.github.com/repos/symfony/symfony-docs');
+
+    $statusCode = $response->getStatusCode();
+    // $statusCode = 200
+    $contentType = $response->getHeaders()['content-type'][0];
+    // $contentType = 'application/json'
+    $content = $response->getContent();
+    // $content = '{"id":521583,"name":"symfony-docs",...}'
+
+If you are consuming APIs, you should use instead the
+:class:`Symfony\\Component\\HttpClient\\ApiClient` class, which includes
+shortcuts and utilities for common operations::
+
+    use Symfony\Component\HttpClient\ApiClient;
+    use Symfony\Component\HttpClient\HttpClient;
+
+    $httpClient = HttpClient::create();
+    $apiClient = new ApiClient($httpClient);
+
+    $response = $apiClient->post('https://api.github.com/gists', [
+        // this is transformed into the proper Basic authorization header
+        'auth' => 'username:password',
+        // this PHP array is encoded as JSON and added to the request body
+        'json' => [
+            'description' => 'Created by Symfony HttpClient',
+            'public' => true,
+            'files' => [
+                'article.txt' => ['content' => 'Lorem Ipsum ...'],
+            ],
+        ],
+    ]);
+
+    // decodes the JSON body of the response into a PHP array
+    $result = $response->asArray();
+    // $result = [
+    //     'id' => '11b5f...023cf9',
+    //     'url' => 'https://api.github.com/gists/11b5f...023cf9',
+    //     ...
+    // ]
+
+Enabling cURL Support
+---------------------
+
+This component supports both the native PHP streams and cURL to make the HTTP
+requests. Although both are interchangeable and provide the same features,
+including concurrent requests, HTTP/2 is only supported when using cURL (and if
+the installed cURL version supports it).
```

Comment:
Which starts from which version? This  could be helpful

Reply:
HTTP/2 support requires:
- PHP `>= 5.5.24` (see php/php-src@1badfd8171119ef70a5d088068b9de88ba668f73)
- liburl `>= 7.38.0`, built with `--with-nghttp2` flag at compile time.

Might also be worth mentioning that the HttpClient component automatically chooses HTTP/2 if the following conditions are met:
- It is supported by cURL,
- The request protocol is `https://`, and
- Another HTTP version has not been explicitly specified in `$options['http_version']`

Reply:
I was too lazy to look for these details, so thanks a lot @zanbaldwin for taking the time to dig them!